src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIPacket.cs

```diff
@@ -249,7 +249,7 @@ public void ReadFromStreamAsync(Stream stream, SNIAsyncCallback callback, bool i
                 options |= TaskContinuationOptions.LongRunning;
             }
 
-            stream.ReadAsync(_data, 0, _capacity).ContinueWith(t =>
+            stream.ReadAsync(_data, 0, _capacity, CancellationToken.None).ContinueWith(t =>
```

Comment:
Use await?

Reply:
@corivera I will send out another PR changing the readAsync piece and making it use async await.